packages/now-cli/src/util/index.js

```diff
@@ -73,361 +70,147 @@ export default class Now extends EventEmitter {
+        if (!files) {
+          files = [];
         }
 
-        const err = new Error(msg);
+        for (const path of paths) {
+          const list = await getFiles(path, {}, opts);
+          files = files.concat(list);
 
-        err.status = res.status;
-        err.retryAfter = 'never';
-
-        return bail(err);
-      }
-
-      // If the deployment domain is missing a cert, bail with the error
-      if (
-        res.status === 400 &&
-        body.error &&
-        body.error.code === 'cert_missing'
-      ) {
-        bail(await responseError(res, null, body));
-      }
-
-      if (
-        res.status === 400 &&
-        body.error &&
-        body.error.code === 'missing_files'
-      ) {
-        return body;
-      }
-
-      if (res.status === 404 && body.error && body.error.code === 'not_found') {
-        return body;
-      }
-
-      if (res.status >= 400 && res.status < 500) {
-        const err = new Error();
-
-        if (body.error) {
-          const { code, unreferencedBuildSpecs } = body.error;
-
-          if (code === 'env_value_invalid_type') {
-            const { key } = body.error;
-            err.message =
-              `The env key ${key} has an invalid type: ${typeof env[key]}. ` +
-              'Please supply a String or a Number (https://err.sh/now/env-value-invalid-type)';
-          } else if (code === 'unreferenced_build_specifications') {
-            const count = unreferencedBuildSpecs.length;
-            const prefix = count === 1 ? 'build' : 'builds';
-
-            err.message =
-              `You defined ${count} ${prefix} that did not match any source files (please ensure they are NOT defined in ${highlight(
-                '.nowignore'
-              )}):` +
-              `\n- ${unreferencedBuildSpecs
-                .map(item => JSON.stringify(item))
-                .join('\n- ')}`;
-          } else {
-            Object.assign(err, body.error);
+          for (const file of list) {
+            relatives[file] = path;
           }
-        } else {
-          err.message = 'Not able to create deployment';
         }
-
-        return bail(err);
       }
+    } else if (type === 'docker') {
+      files = await getDockerFiles(paths[0], nowConfig, opts);
+    }
 
-      if (res.status !== 200) {
-        throw new Error(body.error.message);
+    const uploadStamp = stamp();
+
+    if (isBuilds) {
+      deployment = await processDeployment({
+        now: this,
+        debug,
+        hashes,
+        paths,
+        requestBody,
+        uploadStamp,
+        quiet
+      })
+    } else {
+      // Read `registry.npmjs.org` authToken from .npmrc
+      let authToken;
+
+      if (type === 'npm' && forwardNpm) {
+        authToken =
+          (await readAuthToken(paths[0])) || (await readAuthToken(homedir()));
       }
 
-      for (const [name, value] of res.headers.entries()) {
-        if (name.startsWith('x-now-warning-')) {
-          this._output.warn(value);
```

Comment:
Can you make sure that `now-client` will handle those headers as well, and maybe create an extra warning event for them so that they'll still be printed